File: array/2176_count_equal_and_divisible_pairs_in_an_array/2176_count_equal_and_divisible_pairs_in_an_array.py
class Solution:
    def countPairs(self, nums: List[int], k: int) -> int:
        n = len(nums)
        res = 0
        for i in range(n - 1):
            for j in range(i + 1, n):
                if (i * j) % k == 0 and nums[i] == nums[j]:
                    res += 1
        return res

File: array/2176_count_equal_and_divisible_pairs_in_an_array/2176_count_equal_and_divisible_pairs_in_an_array.go
package main

func countPairs(nums []int, k int) int {
	var res int
	n := len(nums)
	for i := 0; i < n; i++ {
		for j := i + 1; j < n; j++ {
			if nums[i] == nums[j] && (i*j)%k == 0 {
				res++
			}
		}
	}
	return res
}
